api/v1/dois/PKPDoiHandler.php

```diff
@@ -269,28 +275,59 @@ public function edit(SlimRequest $slimRequest, APIResponse $response, array $arg
             return $response->withStatus(403)->withJsonError('api.dois.403.editItemOutOfContext');
         }
 
-        $params = $this->convertStringsToSchema(\PKP\services\PKPSchemaService::SCHEMA_DOI, $slimRequest->getParsedBody());
-        $params['id'] = $doi->getId();
+        $params = $this->convertStringsToSchema(PKPSchemaService::SCHEMA_DOI, $slimRequest->getParsedBody());
 
         $errors = Repo::doi()->validate($doi, $params);
         if (!empty($errors)) {
             return $response->withStatus(400)->withJson($errors);
         }
 
-        Repo::doi()->edit($doi, $params);
+        $pubObjectType = $slimRequest->getParsedBodyParam('pubObjectType');
+        $pubObjectId = $slimRequest->getParsedBodyParam('pubObjectId');
 
-        $doi = Repo::doi()->get($doi->getId());
+        // Default behaviour, only edits DOI
+        if (empty($pubObjectType) && empty($pubObjectId)) {
```

Comment:
This one was a bit messy. It's not always included because this `edit()` function covers two (theoretical) use cases.

1. Being used as a "pure" API endpoint to only edit the DOI with no side effects. (This is the theoretical case, and I'm not sure if this should be allowed as it could lead to unintended consequences). This is the case without including `$pubObjectType` and `$pubObjectId`.
2. Being used to edit the DOIs in the context of their publication object (how it's used in the UI). The code below are the side effects that need to happen when editing a DOI in context. 

```PHP
// Copy DOI object data
$newDoi = clone $doi;
$newDoi->unsetData('id');
$newDoi->setAllData(array_merge($newDoi->getAllData(), ['doi' => $params['doi']]));
$newDoiId = Repo::doi()->add($newDoi);

// Update pubObject with new DOI and remove elsewhere if no longer in use
$pubObjectRepo->edit($pubObject, ['doiId' => $newDoiId]);
if (!Repo::doi()->isAssigned($doi->getId())) {
    Repo::doi()->delete($doi);
}
```

Given this, do you a request should always be required to include a `$pubObjectType` and `$pubObjectId` or is it okay to keep as is?

Reply:
So the 1. case is maybe when editing a DOI manually via the UI/DOI management page?
And the 2. case is e.g. when publishing a new version?
If so maybe to add a comment in the function documentation or here in the code...
And, yes, we can then keep it as is...